backend/queries/admin.js

```diff
@@ -36,13 +36,33 @@ const updateAdmin = async (id, firstName, lastName) => {
 }
 
 const deleteAdmin = async (id) => {
-    return await db.one('DELETE FROM administration WHERE a_id = $1 RETURNING *', id);
+    const deleteQuery = `
+        UPDATE administration 
+            SET deleted = NOW()
+            WHERE a_id = $1
+            RETURNING *
+    `  
+    return await db.one(deleteQuery, id);
+}
+
+const deleteAdminByEmail = async (email, promise) => {
+    const deleteQuery = `
+        UPDATE administration 
+            SET deleted = NOW()
+            WHERE a_email = $1
+            RETURNING *
+    `  
+    if (promise) {
```

Comment:
i couldnt find you calling this with promise in this pull req. what is this conditional for?

Reply:
I wasn't sure to use it or not but all the functions where I added promise as an argument are for simulating the delete cascade on the database, it's meant to push all the async request to the promises array and wait for the array (trigger all the requests at the same time)